### PR TITLE
Connect to upstream services using client ID instead of passing the calling application's client ID.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Connect to upstream services using client ID instead of passing the calling application's client ID.
## What is the intent behind these changes?

Security improvements and making it consistent with other projects.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/config/WebClientConfiguration.kt [hunk 1]
```diff
@@ -53,17 +42,23 @@ class WebClientConfiguration(
...
 
-      next.exchange(
-        ClientRequest.from(request)
-          .header(HttpHeaders.AUTHORIZATION, "Bearer $token")
-          .build(),
-      )
-    }
+  private fun getExchangeStrategies(): ExchangeStrategies {
+    return ExchangeStrategies.builder()
+      .codecs { configurer: ClientCodecConfigurer -> configurer.defaultCodecs().maxInMemorySize(-1) }
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/config/WebClientConfiguration.kt:53]: I see that we've done this in the past. By setting it to -1 we're setting it to unlimited. Can this negatively effect us at all ? More of a question than a request!

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/config/WebClientConfiguration.kt:53]: No, not sure. Copied it from existing code base so not really sure about the settings used and how it impacts.

Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/config/WebClientConfiguration.kt:53]: All other repos use it as well. I won't block the PR but worth us figuring out what this actually means from a security perspective!

**Review by adaviesMOJ — DISMISSED** [review]

**Review by AledWEvans — APPROVED** [review]
LGTM

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**